[Datepicker] custom week start

sdiprizio opened the pull request:
You can now set the first day of the week.

Added a new `@Input` variable `firstDayOfWeek` in `datepicker.ts`. Sunday = 0, so if you want to set Monday as first day of week, just set `firstDayOfWeek="1"` in the ngl-datepicker tag.
Changed `weekday.ts` to reorder weekdays labels accordingly.
Added corresponding test.

Notes:
Did not add consistency check (i.e. if you set firstDayOfWeek = -1234, it should have unexpected bad results).
It's compatible with custom day names (dayNamesShort/dayNamesLong). Those arrays should always start by sunday.

Example of use:

`<ngl-datepicker [(date)]="date" firstDayOfWeek="1" class="slds-box slds-box--x-small"></ngl-datepicker>`

bekos commented:
@sdiprizio I like what I see 👍 I just left some minor comments.
Can you also update `API.md` documenting your change? Thx :-)

sdiprizio commented:
I updated `weekday.ts` to avoid unwanted infinite array growth and updated the loop with your elegant calculation for offset. Added a test to check that `weekdays` size is 7.

In `datepicker.ts` I added a setter on `firstDayOfWeek` to render the calendar on changes. The only side effect I pointed is that it renders the calendar twice on initialization (once when you set `firstDayOfWeek` and once when you set `date`).

Updated the demo with two buttons to test `firstDayOfWeek`.

bekos commented:
@sdiprizio Thank you very much for this quality PR :-)
